dev-haskell/haskeline/haskeline-0.8.1.2.ebuild

```diff
@@ -1,42 +1,37 @@
-# Copyright 1999-2020 Gentoo Authors
+# Copyright 1999-2021 Gentoo Authors
 # Distributed under the terms of the GNU General Public License v2
 
 EAPI=7
 
-# ebuild generated by hackport 0.6.4.9999
+# ebuild generated by hackport 0.6.7.9999
+#hackport: flags: -examples
 
 CABAL_FEATURES="lib profile haddock hoogle hscolour"
 inherit haskell-cabal
+RESTRICT="test" # tests require example executable and don't work correctly
 
 DESCRIPTION="A command-line interface for user input, written in Haskell"
 HOMEPAGE="https://github.com/judah/haskeline"
 SRC_URI="https://hackage.haskell.org/package/${P}/${P}.tar.gz"
 
 LICENSE="BSD"
 SLOT="0/${PV}"
-# keep in sync with ghc-8.10
 KEYWORDS="~amd64 ~x86 ~amd64-linux ~x86-linux"
 IUSE="+terminfo"
 
 RDEPEND=">=dev-haskell/exceptions-0.10:=[profile?] <dev-haskell/exceptions-0.11:=[profile?]
 	>=dev-haskell/stm-2.4:=[profile?] <dev-haskell/stm-2.6:=[profile?]
-	>=dev-lang/ghc-8.0.1:=
+	>=dev-lang/ghc-8.4.3:=
 	terminfo? ( >=dev-haskell/terminfo-0.3.1.3:=[profile?] <dev-haskell/terminfo-0.5:=[profile?] )
 "
 DEPEND="${RDEPEND}
-	>=dev-haskell/cabal-1.24.0.0
+	>=dev-haskell/cabal-2.2.0.1
 "
 
-src_prepare() {
-	default
-
-	cabal_chdeps \
-		'base >=4.9 && < 4.15' 'base >=4.9'
-}
-
 src_configure() {
+	# The 'examples' cabal flag only enables a single executable, which
+	# happens to be only useful for the test
 	haskell-cabal_src_configure \
+		--flag=-examples \
 		$(cabal_flag terminfo terminfo)
 }
-
-CABAL_CORE_LIB_GHC_PV="PM:8.10.2 PM:8.10.3 PM:8.10.4 PM:9999"
```

Comment:
I didn't realize that `haskeline` shipped with GHC. Is updating this going to cause problems? Right now, `ghc-pkg` reports that I have both versions installed:

`data-dir:             /usr/share/x86_64-linux-ghc-8.10.4/haskeline-0.8.0.1`
`data-dir:             /usr/share/haskeline-0.8.1.2/ghc-8.10.4`

It seems to be working fine, but I'm not sure what kinds of complications could arise from this.

Reply:
`purescript` wants `haskeline >= 0.8.1.1 && < 9` which is the reason for the bump.

Reply:
Would using a slot here make sense, so that portage can accurately portray both versions being installed? I'm not sure what the best course of action would be on this one.